Source/Core/DolphinQt/Debugger/BreakpointDialog.cpp

```diff
@@ -113,41 +173,58 @@ void NewBreakpointDialog::CreateWidgets()
 
   auto* layout = new QVBoxLayout;
 
-  layout->addLayout(top_layout);
-  layout->addWidget(m_instruction_box);
-  layout->addWidget(m_memory_bp);
-  layout->addWidget(m_memory_box);
+  switch (m_open_mode)
+  {
+  case OpenMode::New:
+    layout->addLayout(instruction_layout);
+    layout->addLayout(memory_layout);
+    m_instruction_bp->setChecked(true);
+    m_instruction_address->setFocus();
+    break;
+  case OpenMode::EditBreakPoint:
+    layout->addLayout(instruction_layout);
+    m_instruction_bp->setChecked(true);
+    m_instruction_address->setEnabled(false);
+    m_instruction_address->setFocus();
+    break;
+  case OpenMode::EditMemCheck:
+    layout->addLayout(memory_layout);
+    m_cond_help_btn->setVisible(false);
+    m_memory_bp->setChecked(true);
+    m_memory_address_from->setEnabled(false);
+    m_memory_address_to->setFocus();
+    break;
+  }
+
```

Comment:
There's a memory leak here: If you don't add `memory_layout` or `instruction_layout` to the dialog some way, they won't get managed by Qt and thus never get deleted. You should refactor this so the layouts only get created if you actually intend to use them.

Reply:
(alternatively, add them anyway but hide/disable/grey them out, not as elegant but whatever)